platform/jvm/JVMPlatformAssembler.kt

```diff
@@ -0,0 +1,344 @@
+            const val SRC = "src"
+            const val WIX_TOOLSET = "wixtoolset"
+        }
+
+        fun extractAll() {
+            extractJDK()
+            jpackage = findJPackage()
+            if (currentOS == WINDOWS) extractWiXToolset()
+            extractSources()
+        }
+
+        fun extractJDK() {
+            createDirectory(Path.of(JDK))
+            when (currentOS) {
+                MAC, LINUX -> shell.execute(listOf(TAR, "-xf", options.jdkPath, "-C", JDK))
+                WINDOWS -> shell.execute(command = listOf(JAR, "xf", Path.of("..", options.jdkPath).toString()), baseDir = Path.of(JDK))
+            }
+        }
+
+        fun extractWiXToolset() {
+            createDirectory(Path.of(WIX_TOOLSET))
+            shell.execute(command = listOf(JAR, "xf", Path.of("..", options.windowsWiXToolsetPath).toString()), baseDir = Path.of(WIX_TOOLSET))
+        }
+
+        fun findJPackage(): File {
+            val name = if (currentOS == WINDOWS) JPACKAGE_EXE else JPACKAGE
+            return File(JDK).listFilesRecursively().firstOrNull { it.name == name }
+                ?: throw IllegalStateException("Could not locate '$name' in the provided JDK")
+        }
+
+        fun extractSources() {
+            val tempDir = "src-temp"
+            createDirectory(Path.of(tempDir))
+            shell.execute(command = listOf(JAR, "xf", Path.of("..", options.sourceFilename).toString()), baseDir = Path.of(tempDir))
+            // Emulate the behaviour of `tar -xf --strip-components=1`
+            val files = File(tempDir).listFiles()
+            assert(files!!.size == 1)
+            assert(files[0].isDirectory)
+            Files.move(files[0].toPath(), srcDir.toPath())
+            File(tempDir).deleteRecursively()
+        }
+    }
+
+    private sealed class PlatformImageBuilder {
+        protected lateinit var version: String
+        protected val shortVersion: String; get() = version.split("-")[0] // e.g: 2.0.0-alpha5 -> 2.0.0
+
+        fun build() {
+            version = readVersionFile()
+            beforePack()
+            pack()
+            setPackageFilename()
+            afterPack()
+        }
+
+        private fun readVersionFile(): String {
+            return inputFiles.version.readLines()[0]
+        }
+
+        protected open fun beforePack() {}
+
+        protected open fun pack() {
+            shell.execute(
+                command = listOf(inputFiles.jpackage.path) + packArgsCommon() + packArgsPlatform(),
+                env = packEnv()
+            )
+        }
+
+        private fun setPackageFilename() {
+            // TODO: document the reasons behind this method
+            distDir.listFiles()!![0].let {
+                it.renameTo(
+                    File(it.path.replace(options.image.name, options.image.filename).replace(shortVersion, version))
+                )
+            }
+        }
+
+        protected open fun packEnv(): Map<String, String> {
```

Comment:
`jpackage` requires a `PATH` environment variable override to work in Windows - but on most operating systems, we don't need to set any env vars.